File: social/src/main/java/com/example/social/dto/PostRelation.java
package com.example.social.dto;

import com.example.social.document.CommentDocument;
import com.example.social.document.FavoriteDocument;
import com.example.social.document.PostDocument;
import com.example.social.document.UserDocument;
import lombok.Data;

import java.util.List;

@Data
public class PostRelation {
    private UserDocument user;
    private PostDocument post;
    private List<FavoriteDocument> favorites;
    private List<CommentDocument> comments;
}

File: socials-mobile/lib/models/post_relation.dart
import 'package:socials/models/post.dart';
import 'package:socials/models/user.dart';

import 'favorite.dart';

class PostRelation {
  User? user;
  Post? post;
  List<Favorite>? favorites;

  PostRelation(this.user, this.post, this.favorites);
}